Kubernetes

SamPixelYork opened the pull request:
Files for running application under Kubernetes. See README.md for instructions

SamPixelYork committed:
Initial setup for Kubernetes

SamPixelYork committed:
Added the rest of the k8 files

SamPixelYork committed:
Pod deployments initially run, then shutdown

SamPixelYork committed:
Removed docker image file

Creating file to test git commit, push, and pull

SamPixelYork committed:
5 pods running - using local images

SamPixelYork committed:
Added a readme on starting the Kubernetes environment

SamPixelYork committed:
5 pods running - using local images

SamPixelYork committed:
Fix typo in running command

SamPixelYork committed:
All bar x and frred pods running

SamPixelYork committed:
All pods running

SamPixelYork committed:
Small correction in x.yaml volume mount

SamPixelYork committed:
Added script to start kubernetes

SamPixelYork committed:
Add metrics command to ensure dashboard runs

SamPixelYork committed:
Extra line removed from script

SamPixelYork committed:
Update README to include script details

Heroj04 commented:
A couple things I've noticed in my testing. Also noted in the Kubernetes.docx file.

1. Running the prod environment before running kubernetes will cause it to not run because the file hosting doesnt work.  
Deleting the orphan prod containers will solve this issue.  
This isnt really a major problem because we're moving away from docker compose anyway so the old containers will be removed regardless.
2. It seems like using `ctrl+c` after the running `kubernetes.sh` will kill the file hosting services.  
This would cuase the system to not be able to save or read any data after it is closed and instead just running everything from memory.
3. The required ports are not exposed on the host system.  
The ports are exposed from the pods to the main minikube container but they are not exposed from minikube to the host.  
The required ports from the docker compose system are: 
- `mq`
  - `1884:1883`
  - `15672:15672`
  - `15692:15692`
- `restapi`
  - `5687:5687`
- `ttn_webhook`
  - `5688:5688`
- `website`
  - `5000:5000`

SamPixelYork committed:
Using configmap to mount python files

Heroj04 committed:
Kubernetes Cleanup
- Remove Test Files
- Fix Services (only where needed and some exposed via NodePorts)
- Volumes are mounted via HostPath

Heroj04 committed:
Fix configmap reference

Heroj04 committed:
Remove unnecessary files

Heroj04 committed:
Use a namespace

Heroj04 committed:
Move microservice configurations to subdirectory

Heroj04 committed:
Update kubernetes script to allow stop and start
- Remove minikube reliance

Heroj04 committed:
Update kubernetes README

Heroj04 committed:
Revert Changes Made in Kubernetes Testing

Heroj04 committed:
Move and rename kubernetes documentation

Heroj04 committed:
Add Databolt Volume

Heroj04 committed:
fix mq volume location

Heroj04 committed:
fix ttn volume location

Heroj04 committed:
Fix docker compose indenting

Heroj04 committed:
readme touchups

Heroj04 committed:
Change imagePullPolicy to allow using online images

Heroj04 committed:
Add information about locally built images

Heroj04 committed:
Remove latest tag from local images

Heroj04 committed:
Merge branch 'Development' into feature/kubernetes

Heroj04 committed:
Add status command

Heroj04 committed:
Add deployment for x test container

Heroj04 committed:
Add exposed nodeports as seperate services

Heroj04 committed:
Add eagleio api token to configmap

Heroj04 committed:
Add installation instructions
